llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -883,6 +883,114 @@ static bool removeDeviceGlobalFromCompilerUsed(Module &M) {
   return true;
 }
 
+// Add extra red zone to each image scope device globals if the module has been
+// instrumented by sanitizer pass. And record their infomation like size, red
+// zone size, beginning address.
+static bool instrumentDeviceGlobal(Module &M) {
+  if (!M.getNamedGlobal("__DeviceSanitizerReportMem"))
+    return false;
```

Comment:
If this checking method is stable, can you help to change "isModuleUsingAsan" function, and use it here?

Reply:
Done